resources/providers/azurelib/inventory/susbcriptions_provider.go

```diff
@@ -0,0 +1,90 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package inventory
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/samber/lo"
+
+	"github.com/elastic/cloudbeat/resources/utils/pointers"
+)
+
+type subscriptionAzureClientWrapper struct {
+	AssetLocations func(ctx context.Context, subID string, clientOptions *arm.ClientOptions, options *armsubscriptions.ClientListLocationsOptions) ([]armsubscriptions.ClientListLocationsResponse, error)
+}
+
+type SubscriptionProviderAPI interface {
+	ListLocations(ctx context.Context, subID string) ([]AzureAsset, error)
+}
```

Comment:
Regarding the name: Isn't this a "location provider"? it doesn't provide any subscriptions. The subscription part is only relevant because the implementation (azure) returns a list of locations under the subscription API for some reason.

Reply:
Hmmmmm, I followed the pattern I was following on the other providers. I named a subscription provider because it uses the subscription resource of azure. 

The same way mysql only return configurations but is called mysql provider. I could rename to locations, but it would narrow the scope of this provider quite a lot and differ from the others